java/build-jar-multiplatform.sh

```diff
@@ -0,0 +1,146 @@
+    cp -f "$WORKSPACE_DIR"/java/"$module"/target/*jar "$JAR_DIR"
+  done
+  for module in "${RAY_STREAMING_JAVA_MODULES[@]}"; do
+    cp -f "$WORKSPACE_DIR"/streaming/java/"$module"/target/*jar "$JAR_DIR"
+  done
+  # ray runtime jar and streaming runtime jar are in a dir specifed by maven-jar-plugin
+  cp -f "$WORKSPACE_DIR"/build/java/ray*.jar "$JAR_DIR"
+  cp -f "$WORKSPACE_DIR"/streaming/build/java/streaming*.jar "$JAR_DIR"
+}
+
+# This function assuem all dependencies are installed already.
+build_jars_linux() {
+  build_jars linux
+}
+
+# This function assuem all dependencies are installed already.
+build_jars_darwin() {
+  build_jars darwin
+}
+
+build_jars_multiplatform() {
+  download_jars
+  prepare_native
+  build_jars multiplatform false
+}
+
+# download linux/darwin ray-runtime-$version.jar from s3
+download_jars() {
+   local version
+   # ray jar version, ex: 0.1-SNAPSHORT
+   version=$(python -c "import xml.etree.ElementTree as ET;  r = ET.parse('pom.xml').getroot(); print(r.find(r.tag.replace('project', 'version')).text);" | tail -n 1)
+  wait_time=0
+  sleep_time_units=1
+  for os in 'darwin' 'linux'; do
+    url=https://ray-wheels.s3-us-west-2.amazonaws.com/jars/"$TRAVIS_BRANCH/$TRAVIS_COMMIT/$os/ray-runtime-$version.jar"
+    dest_file="$JAR_BASE_DIR/$os/ray-runtime-$version.jar"
+    echo "Jar url: $url"
+    echo "Jar dest_file: $dest_file"
+    while true; do
+      if ! wget -q "$url" -O "$dest_file">/dev/; then
+        echo "Waiting $url to be ready for $wait_time seconds..."
+        sleep $sleep_time_units
+        wait_time=$((wait_time + sleep_time_units))
+        if [[ wait_time == $((60 * 30)) ]]; then
+          echo "Download $url timeout"
+          exit -1
+        fi
+      else
+        echo "Download $url to $dest_file succeed"
+        break
+      fi
+    done
+  done
+}
+
+# prepare native binaries and libraries.
+prepare_native() {
+  for os in 'darwin' 'linux'; do
+    cd "$JAR_BASE_DIR"/"$os"
+    jar xf "$(ls ray-runtime*.jar | grep -v sources | grep -v grep)" "$os"
+    local native_dir="$WORKSPACE_DIR"/java/runtime/native_dependencies/native/"$os"
+    mkdir -p "$native_dir"
+    rm -rf "$native_dir"
+    jar xf "$(ls ray-runtime*.jar | grep -v sources | grep -v grep)" "native/$os"
+    mv "native/$os" "$native_dir"
+    local native_dir="$WORKSPACE_DIR"/streaming/java/streaming-runtime/native_dependencies/native/"$os"
+    mkdir -p "$native_dir"
+    rm -rf "$native_dir"
+    jar xf "$(ls streaming-runtime*.jar | grep -v sources | grep -v grep)" "native/$os"
+    mv "native/$os" "$native_dir"
+  done
+}
+
+# This function assume all multiplatform binaries are prepared already.
+deploy_jars() {
+  echo "Start deploying jars"
+  cd "$WORKSPACE_DIR"/java
+  mvn -T16 deploy -Dmaven.test.skip=true -Dcheckstyle.skip
+  cd "$WORKSPACE_DIR"/streaming/java
+  mvn -T16 deploy -Dmaven.test.skip=true -Dcheckstyle.skip
+  echo "Finished deploying jars"
+}
+
+case $1 in
+linux)
```

Comment:
```suggestion
linux*)
```
You need the asterisk.

Reply:
Why `linux*` here?

Reply:
The input parameter is `linux` without suffix

Reply:
Oh I assumed it was `${OSTYPE}`. If the input is literally "linux" then never mind.